web/pdf_thumbnail_view.js

```diff
@@ -156,296 +148,282 @@ var PDFThumbnailView = (function PDFThumbnailViewClosure() {
-      this.viewport = this.viewport.clone({
-        scale: 1,
-        rotation: totalRotation,
-      });
-      this.reset();
-    },
+  update(rotation) {
+    if (typeof rotation !== 'undefined') {
+      this.rotation = rotation;
+    }
+    let totalRotation = (this.rotation + this.pdfPageRotate) % 360;
+    this.viewport = this.viewport.clone({
+      scale: 1,
+      rotation: totalRotation,
+    });
+    this.reset();
+  }
 
-    cancelRendering: function PDFThumbnailView_cancelRendering() {
-      if (this.renderTask) {
-        this.renderTask.cancel();
-        this.renderTask = null;
-      }
-      this.renderingState = RenderingStates.INITIAL;
-      this.resume = null;
-    },
+  cancelRendering() {
+    if (this.renderTask) {
+      this.renderTask.cancel();
+      this.renderTask = null;
+    }
+    this.renderingState = RenderingStates.INITIAL;
+    this.resume = null;
+  }
 
-    /**
-     * @private
-     */
-    _getPageDrawContext:
-        function PDFThumbnailView_getPageDrawContext(noCtxScale) {
-      var canvas = document.createElement('canvas');
-      // Keep the no-thumbnail outline visible, i.e. `data-loaded === false`,
-      // until rendering/image conversion is complete, to avoid display issues.
-      this.canvas = canvas;
+  /**
+   * @private
+   */
+  _getPageDrawContext(noCtxScale = false) {
+    let canvas = document.createElement('canvas');
+    // Keep the no-thumbnail outline visible, i.e. `data-loaded === false`,
+    // until rendering/image conversion is complete, to avoid display issues.
+    this.canvas = canvas;
+
+    if (typeof PDFJSDev === 'undefined' ||
+        PDFJSDev.test('MOZCENTRAL || FIREFOX || GENERIC')) {
+      canvas.mozOpaque = true;
+    }
+    let ctx = canvas.getContext('2d', { alpha: false, });
+    let outputScale = getOutputScale(ctx);
 
-      if (typeof PDFJSDev === 'undefined' ||
-          PDFJSDev.test('MOZCENTRAL || FIREFOX || GENERIC')) {
-        canvas.mozOpaque = true;
-      }
-      var ctx = canvas.getContext('2d', { alpha: false, });
-      var outputScale = getOutputScale(ctx);
+    canvas.width = (this.canvasWidth * outputScale.sx) | 0;
+    canvas.height = (this.canvasHeight * outputScale.sy) | 0;
+    canvas.style.width = this.canvasWidth + 'px';
+    canvas.style.height = this.canvasHeight + 'px';
 
-      canvas.width = (this.canvasWidth * outputScale.sx) | 0;
-      canvas.height = (this.canvasHeight * outputScale.sy) | 0;
-      canvas.style.width = this.canvasWidth + 'px';
-      canvas.style.height = this.canvasHeight + 'px';
+    if (!noCtxScale && outputScale.scaled) {
+      ctx.scale(outputScale.sx, outputScale.sy);
+    }
+    return ctx;
+  }
 
-      if (!noCtxScale && outputScale.scaled) {
-        ctx.scale(outputScale.sx, outputScale.sy);
-      }
-      return ctx;
-    },
+  /**
+   * @private
+   */
+  _convertCanvasToImage() {
+    if (!this.canvas || this.renderingState !== RenderingStates.FINISHED) {
```

Comment:
Given that these two conditions are quite different, I'm wondering if having two different `if`s (as in the current code) actually aid readability somewhat here!?